app/knight_class.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.equipment = {"armour": [], "weapon": "", "potion": ""}
+        print(f"Knight {name} arrived at the championship!")
+
+    def add_armours(self, armours_list: list) -> None:
+        for item in armours_list:
+            if "part" in item.keys() and "protection" in item.keys():
+                self.equipment["armour"].append(item["part"])
+                self.protection += item["protection"]
```

Comment:
Consider adding validation to check if `armours_list` is a list of dictionaries with the required keys ('part' and 'protection'). This will prevent errors if the input is not as expected.